src/stores/SpaceStore.tsx

```diff
@@ -120,7 +120,19 @@ export class SpaceStoreClass extends AsyncStoreWithClient<IState> {
      * should not be done when the space switch is done implicitly due to another event like switching room.
      */
     public async setActiveSpace(space: Room | null, contextSwitch = true) {
-        if (space === this.activeSpace || (space && !space?.isSpaceRoom())) return;
+        if (space && !space?.isSpaceRoom()) return;
+        if (space && space === this.activeSpace) {
+            const notificationState = this.getNotificationState(space.roomId);
+            if (notificationState.count) {
+                const roomId = notificationState.getRoomWithMaxNotifications();
+                defaultDispatcher.dispatch({
+                    action: "view_room",
+                    room_id: roomId,
+                    context_switch: true,
+                });
+            }
+            return;
+        } else if (space === this.activeSpace) return;
```

Comment:
Sorry, if I am missing something here. Couldn't we also select the room with max notification from the home space here?

Reply:
Yes, we can do that. I thought that the behaviour wasn't intended for the home space. I'll modify this soon.

Reply:
It would make sense to me but as usual it's up to the design team